Researchers are curious to know if four different engine oils lead to different mileage of cars. To test this, they measured the mileage of 5 cars using four different engine oils. Since each car’s mileage is measured by applying each of the four-engine oils one by one they use a repeated-measures ANOVA to check if the mean reaction time differs between engine oils.



Hypotheses:

null hypothesis (H0): µ1 = µ2 = µ3
alternative hypothesis: (Ha): at least one population mean differs from the rest

In [1]:
# Import library
import numpy as np
import pandas as pd
from statsmodels.stats.anova import AnovaRM

# Create the data
dataframe = pd.DataFrame({'Cars': np.repeat([1, 2, 3, 4, 5], 4),
                          'Oil': np.tile([1, 2, 3, 4], 5),
                          'Mileage': [36, 38, 30, 29,
                                      34, 38, 30, 29,
                                      34, 28, 38, 32,
                                      38, 34, 20, 44,
                                      26, 28, 34, 50]})

# Conduct the repeated measures ANOVA
print(AnovaRM(data=dataframe, depvar='Mileage',
              subject='Cars', within=['Oil']).fit())

              Anova
    F Value Num DF  Den DF Pr > F
---------------------------------
Oil  0.5679 3.0000 12.0000 0.6466



Result Analysis

In this example, the F-test statistic is 0.5679 and the corresponding p-value is 0.6466.
Since this p-value is not less than 0.05, we cannot reject the null hypothesis and conclude that there is no statistically significant difference in the average response times between the four engine oils.


Outliers:
There were not outliers found in the plots and data.

Normality: The residuals (errors) from the ANOVA model are assumed to be normally distributed. This assumption can be checked using normality tests, such as the Shapiro-Wilk test. The data were normally distributed.

Homoscedasticity: The variances of the residuals are assumed to be equal across all groups (engine oils). This assumption can be checked using Levene's test for equality of variances. According to the data, there was homogeneity of variances.

Sphericity: The error covariance matrix is assumed to be spherical, meaning that the variances of the residuals are equal for all pairs of repeated measures and the covariances between the residuals are all zero. This assumption can be checked using Mauchly's test of sphericity.